### PR TITLE
rsc: Create configurable no-op rsc runner

### PR DESCRIPTION
**Pull request by V-FEXrt**
Creates a rsc runner that just falls through to the underlying runner. The RSC runner can be configured with the ENV var `WAKE_REMOTE_CACHE` with the format `domain:port:auth` or `domain:port:` for no auth.

Example: `WAKE_REMOTE_CACHE=localhost:3001:MyKey ./bin/wake -x Unit`

This PR will race with https://github.com/sifive/wake/pull/1519

### CODE REVIEW

File: share/wake/lib/system/runner.wake [hunk 1]
```diff
@@ -148,17 +148,34 @@ export def fuseRunner: Runner =
...
+        else fuseRunner
+
+        # The fuseRunner does not actully mount over / and instead uses the
+        # the host root as the sandbox root. This means that wakeroot will
+        # change from depending on where wake is being run from. As a hack
+        # to work around this we abuse the fact that the fuseRunner only
+        # works in relative paths to make different runs consistent. Ideally
+        # you'd have a more complex sandbox that kept the wakeroot at a
+        # consistent place across runs.
+        mkJobCacheRunner (\_ Pass "") "/workspace" fuseRunner
```
Comment (colinschmidt) [inline share/wake/lib/system/runner.wake:163]: Nit, but sometimes these nested requires are easier to read as a match.

File: share/wake/lib/system/runner.wake [hunk 2]
```diff
@@ -148,17 +148,34 @@ export def fuseRunner: Runner =
...
+        panic "Remote cache config was set with incorrect format. Saw: '{config}'. Expected 'domain:port:auth' (auth may be omitted)"
+
+    require Some (port) =
+        portStr
+        | int
+    else panic "Remote cache config was set with non-integer port. Saw: {portStr}"
+
+    def auth = if authStr ==* "" then None else Some authStr
+
+    rscRunner (RemoteCacheApi domain port auth)
```
Comment (colinschmidt) [inline share/wake/lib/system/runner.wake:178]: It feels like this should maybe be another function that lives in remote_cache_api.wake
`def makeRemoteCacheAPI configKey: RemoteCacheAPI`

File: share/wake/lib/system/runner.wake (hunk 1 above)
Comment (V-FEXrt) [reply share/wake/lib/system/runner.wake:163]: This is another example where `when` would be _so much_ nicer

```
def defaultRunner =
    when Some config = getenv "WAKE_REMOTE_CACHE"
        mkRscRunner ...
    when Some _ = getenv "WAKE_LOCAL_JOB_CACHE"
        mkJobCacheRunner ...
    fuseRunner
```

I'm still partial to the decreased nesting of `require` over `match` here though

**Review by colinschmidt — APPROVED** [review]
LGTM

**Pull request merged**